File: packages/android/src/main/java/com/kozmos/components/RoutingInputGroup/RoutingInputGroup.figma.kt
package com.kozmos.components.routinginputgroup

import androidx.compose.runtime.Composable
import com.figma.code.connect.Figma
import com.figma.code.connect.FigmaConnect
import com.figma.code.connect.FigmaProperty
import com.figma.code.connect.FigmaType

@FigmaConnect("https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=1351-8315")
class KozmosRoutingInputGroupConnect {
    @FigmaProperty(FigmaType.Text, "Point Label Text")
    val pointLabel: String = "Start"

    @Composable
    fun ComponentExample() {
        KozmosRoutingInputGroup(
            points = listOf(
                KozmosRoutePoint(id = "origin", value = "", placeholder = pointLabel),
                KozmosRoutePoint(id = "destination", value = "", placeholder = "Destination")
            ),
            onPointChange = { _, _ -> },
            onSwap = {}
        )
    }
}

File: packages/ios/Sources/Components/RoutingInputGroup/RoutingInputGroup.figma.swift
import SwiftUI
import Figma

struct KozmosRoutingInputGroupConnect: FigmaConnect {
    let component = KozmosRoutingInputGroup.self
    let figmaNodeUrl = "https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=1351-8315"

    @FigmaString("Point Label Text")
    var pointLabel: String = "Start"

    // Content is the length of points — two for origin and destination, three
    // once a stop is added — rather than a parameter.
    var body: some View {
        KozmosRoutingInputGroup(
            points: [
                KozmosRoutePoint(id: "origin", value: "", placeholder: self.pointLabel),
                KozmosRoutePoint(id: "destination", value: "", placeholder: "Destination")
            ],
            onPointChange: { _, _ in },
            onSwap: {}
        )
    }
}
